cmd/konvoy-image/cmd/vsphere.go

```diff
@@ -0,0 +1,146 @@
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+
+	"github.com/mesosphere/konvoy-image-builder/pkg/app"
+)
+
+var (
+	vsphereExample = "vsphere --datacenter dc1 --cluster zone1 --datastore nfs-store1 " +
+		"--network public --template=d2iq-base-templates/d2iq-base-CentOS-7.9 " +
+		"images/ami/centos-79.yaml"
+	vsphereUse = "vsphere <image.yaml>"
+)
+
+func NewVSphereBuildCmd() *cobra.Command {
+	flags := &buildCLIFlags{}
+	cmd := &cobra.Command{
+		Use:     vsphereUse,
+		Short:   "build and provision vsphere images",
+		Example: vsphereExample,
+		Args:    cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			runBuild(args[0], flags)
+		},
+	}
+
+	initBuildVSphereFlags(cmd.Flags(), flags)
+	return cmd
+}
+
+func NewVSphereGenerateCmd() *cobra.Command {
+	flags := &generateCLIFlags{}
+	cmd := &cobra.Command{
+		Use:     vsphereUse,
+		Short:   "generate files relating to building vsphere images",
+		Example: vsphereExample,
+		Args:    cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			runGenerate(args[0], flags)
+		},
+	}
+
+	initGenerateVSphereFlags(cmd.Flags(), flags)
+	return cmd
+}
+
+func initBuildVSphereFlags(fs *flag.FlagSet, buildFlags *buildCLIFlags) {
+	initGenerateArgs(fs, &buildFlags.generateCLIFlags)
+	initVSphereArgs(fs, &buildFlags.generateCLIFlags)
+
+	addBuildArgs(fs, buildFlags)
+}
+
+func initGenerateVSphereFlags(fs *flag.FlagSet, generateFlags *generateCLIFlags) {
+	initGenerateArgs(fs, generateFlags)
+	initVSphereArgs(fs, generateFlags)
+}
+
+func initVSphereArgs(fs *flag.FlagSet, gFlags *generateCLIFlags) {
+	gFlags.userArgs.VSphere = &app.VSphereArgs{}
+	addVSphereArgs(fs, gFlags.userArgs.VSphere)
+}
+
+func addVSphereArgs(fs *flag.FlagSet, vsphereArgs *app.VSphereArgs) {
+	fs.StringVar(
+		&vsphereArgs.Template,
+		"template",
+		"",
+		"Base template to be used. Can include folder. <templatename> or <folder>/<templatename>."+
+			"Required value: you can pass the template name through an override file, or image definition file.",
+	)
+	fs.StringVar(
+		&vsphereArgs.Datacenter,
+		"datacenter",
+		"",
+		"The vSphere datacenter."+
+			"Required value: you can pass the datacenter name through an override file, or image definition file.",
```

Comment:
Change to `"Required value: you can pass the datacenter name through an override file or image definition file."` here and every other line that has the verbiage. 

Add a space character anywhere it says `Required value`

--host and --template also need spaces per @JohnB64-JBV. Review the rest of the code to double-check.

Reply:
Fixed!